portfolio/src/main/webapp/charts.js

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawPoll);
+
+// Map that holds all polls that can be voted on.
+const polls = new Map([
+  [
+    'catsDogs', {
+      title: 'Do you Prefer Cats or Dogs?',
+      choices: ['Cats', 'Dogs'],
+    }
+  ],
+  [
+    'iceCream', {
+      title: 'What is the Best Ice Cream Flavor?',
+      choices: [
```

Comment:
Not something I think you should bother changing but just as a followup to the discussion we had earlier, this is an example of something that can be misused when it's client-side. Users could add their own options to the poll and then post those, which is probably not your actual intent here.

The safer way of doing this is to store all the polls and their options on the server and have a JSON endpoint that returns them to JS. Then the POST request to /vote-result can validate that the option the user selected is actually one of the choices.

Reply:
That is a very good point! Something that I will keep in mind as I move on.